### ISSUE TITLE
Nested arrow functions with await will use up a lot of memory

### ISSUE THREAD

**Issue by rol1510** [quick-lint/quick-lint-js#827]
I don't know if this is important or is worth fixing, but I did some fuzzing and found that if you keep repeating the pattern `await identifier arrow-operator` often enough, it will take forever to lint and the memory usage will explode.

```javascript
// tmp.js
await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await v1 => await
```
or
```javascript
// tmp2.js
await await => await await => await await => await await => await await => await await => await await => await await => await await => await await => await await => await await => await await => await await => await await => await await => await await
```

	$ time ./build/quick-lint-js tmp.js
	tmp.js:2:1: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:13: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:25: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:37: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:49: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:61: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:73: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:85: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:97: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:109: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:121: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:133: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:145: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:157: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:169: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:181: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:193: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:205: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:217: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:229: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
	tmp.js:2:241: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]

	real	0m6,918s
	user	0m2,230s
	sys	0m4,684s

The examples will use more than 10GB of RAM, and if I add one or two more `await v1 =>` it looks like this:

<img width="410" alt="image" src="https://user-images.githubusercontent.com/29011024/186012117-360353b5-f3e3-4b27-9eb7-0b7b398af572.png">

But on the other hand, if you remove a few awaits it's almost not detectable.

There also can be some other stuff in between. Same result.
```js
=== % await ! await v1 => ! await v1 => ! - ! await v1 => ! await v1 => ! - await v1 => await v1 => ++ await v1 => ! await v1 => ++ await === % await ! await v1 => ! await v1 => ! - ! await v1 => ! await v1 => ! - await v1 => await v1 => ++ await v1 => ! await v1 => ++ await v1 => ! await v1 => . v1 => ! await v1 => . = switch => await await 
```
This one just goes on for ever spitting out the same error, but doesn't hog any memory:

```js
await => await await => await => await await v1 => ! await v1 => await v1 => ! await v1 => await v1 => => { - v1 let v1 => { - v1 let v1 v1 : typeof await === === + await ! await v1 => ! await v1 => ! await v1 => await v1 => ! await v1 => await v1 => => { - v1 let v1 => { - v1 let v1 === + await ! await v1 => ! await v1 => ! await v1 => await v1 => ! await v1 => await v1 => => { - v1 let v1 => { - v1 : typeof v3 | v3 . case 
```

I ran it in an Ubuntu 22.04 VM on x86. Release and Debug builds have the same outcomes.
Looks like the browser version is just crashing.

**Comment by strager** [comment]
Thanks for reporting! It looks like there are several root causes. I'll work on a fix.

**Comment by strager** [comment]
I fixed the error message spam in commit 4ebe1677198dcfb2cff7281d1ff69be2407f2f0b.

**Comment by strager** [comment]
Fixed the performance issue in commit 0a463a804287e02d9e88204800915854a0660c2c.

**Comment by strager** [comment]
Fixes released in version 2.9.0. https://quick-lint-js.com/releases/
